src/core/equals.spec.js

```diff
@@ -22,19 +31,31 @@ test('equals primatives', t => {
   t.end()
 })
 
-test('equals adts', t => {
-  const value = 'value'
-  const spy = sinon.spy(() => value)
+test('equals Setoid', t => {
```

Comment:
I think we should add some test related to the `this` context. Since someone can remove the bind and the test will still work.

```js
const b = {
  prop: 'yay',
  equals(arg) {
    this.foo === arg
  }
};
```

Reply:
I do not understand what you mean.
Does ln 42 not cover this?

Reply:
Oh, I mean like use some object that has a method `equals` and inside it use some value that is a property of the object. Like the variable `b` that I mention. It has a property `prop` and inside the `b.equals` method we are checking against that property using `this.prop`. if we remove the `bind` in the implementation of the `equals`, `this` will be undefined.

Reply:
Oooh, you are right, sorry, did not get it.